### DISCUSSION TITLE
Auto-format if there is a config

### DISCUSSION THREAD

**Discussion by Johan-Mi** [helix-editor/helix#8651] [Ideas]
If I am working on a C project that has a Clang-format configuration, I would like Helix to enable auto-format. If I, however, edit a C file in a directory that *doesn't* have a formatter configuration, then auto-format would mess up the formatting. Would it be possible to implement some way for formatters to specify a file to check for to see if auto-format should be used or not?

**Comment by dead10ck** [comment]
No, this is not possible, and I don't think there would be much appetite for such a feature, as it's pretty niche.

However, you can make project-specific Helix settings by making a `.helix/config.toml` that enables or disables auto formatting for that project.
